docs/configuration/index.md

```diff
@@ -750,6 +750,7 @@ Prior to version 0.13.0, Druid string columns treated `''` and `null` values as
 |Property|Description|Default|
 |---|---|---|
 |`druid.generic.useDefaultValueForNull`|When set to `true`, `null` values will be stored as `''` for string columns and `0` for numeric columns. Set to `false` to store and query data in SQL compatible mode.|`true`|
+|`druid.generic.ignoreNullsForStringCardinality`|When set to `true`, `null` values will be ignored for cardinality aggregation over string columns. Set to `false` to include `null` values while estimating cardinality of string columns|`false`|
```

Comment:
My thoughts here are that this sort of feature flag is better placed as a QueryContext flag since it is meant to only impact query results. We can then take advantage of #10208 to allow operators to set a default for all queries on the cluster.

Reply:
I think query context alone isn't enough, since I think it might make sense to allow ingestion to behave this way as well. It would be a bit strange if you used the cardinality agg at ingest time to discard some string and get a count of 2, but then if you had left the raw data you could query as 1 or 2 depending on the setting.

I think @jihoonson suggestion as a parameter for cardinality agg would be fine if we think this should be something that is dynamic, but that alone would need some additional mechanism to tell SQL to set it when making the count distinct probably. I don't personally have a very strong preference, the static config has some attractiveness since it is less disruptive, but it probably isn't much trouble to pass it through everywhere either.

Reply:
To add a bit more to this, i don't think aggs have easy access to query context, so query context could potentially be the mechanism which SQL sets the property on the cardinality aggregator at query time, if we also add it to the cardinality aggregator

Reply:
> since I think it might make sense to allow ingestion to behave this way as well.
Good point. I had not considered the cardinality aggregator being used at ingestion time.

Reply:
```suggestion
|`druid.generic.ignoreNullsForStringCardinality`|When set to `true`, `null` values will be ignored for the built-in cardinality aggregator over string columns. Set to `false` to include `null` values while estimating cardinality of string columns|`false`|
```